R/calcUvalueParameters.R

```diff
@@ -0,0 +1,567 @@
+
+    # we have some 0 entries due to insufficient resolution for very small islands
+    filter(.data$value != 0)
+
+
+  # Aggregate HDD/CDDS to MessageIX regions
+  #TODO: replace this by calcOutput("HDDCDD", aggregate = TRUE, ...)   #nolint: todo_comment_linter
+  pop <- pop %>%
+    filter(.data$variable == "pop_SSP2") %>%
+    mutate(variable = sub("_SSP2", "", .data$variable)) %>%
+    select("region", "period", "variable", "value")
+
+  hddcddMessage <- hddcdd %>%
+    rbind(pop) %>%
+    mutate(unit = NA) %>%
+    aggregate_map(mapping = regionmapMessage %>%
+                    semi_join(hddcdd, by = "region"),
+                  by = "region",
+                  subset2agg = c("HDD", "CDD"),
+                  weights = "pop",
+                  forceAggregation = TRUE) %>%
+    select(-"unit")
+
+  # Full historical HDD/CDD data for all relevant regions
+  hddcddEstimate <- hddcdd %>%
+    filter(!.data$region %in% unique(regionmapMessage$regionTarget)) %>%
+    rbind(hddcddMessage) %>%
+    pivot_wider(names_from = "variable", values_from = "value")
+
+
+  ### GDP per capita ====
+
+  # Aggregate GDP/pop to MessageIX regions
+  gdppop <- gdppop %>%
+    mutate(variable = "gdppop") %>%
+    filter(.data$period <= endOfHistory) %>%
+    select("region", "period", "variable", "value")
+
+  gdppopMessage <- gdppop %>%
+    rbind(pop) %>%
+    mutate(unit = NA) %>%
+    aggregate_map(mapping = regionmapMessage %>%
+                    semi_join(gdppop, by = "region"),
+                  by = "region",
+                  subset2agg = "gdppop",
+                  weights = "pop",
+                  forceAggregation = TRUE) %>%
+    select(-"unit")
+
+  # Full historical GDP/pop data for all relevant regions
+  gdppopEstimate <- gdppop %>%
+    filter(!.data$region %in% unique(regionmapMessage$regionTarget)) %>%
+    rbind(gdppopMessage) %>%
+    pivot_wider(names_from = "variable", values_from = "value")
+
+
+  ### Full data sets ====
+
+  # Gather full historical data for estimates on u-values, HDD/CDDs and GDP/pop
+  dataEstimate <- uvaluesHotmaps %>%
+    rbind(uvaluesEUBuildObs,
+          uvaluesMessageAgg) %>%
+    right_join(hddcddEstimate, by = c("region", "period"),
+               relationship = "many-to-many") %>%
+    left_join(gdppopEstimate, by = c("region", "period")) %>%
+    filter(!is.na(.data$uvalue))
+
+
+  # Gather full historical data for predictions
+  dataHist <- gdppop %>%
+    semi_join(hddcdd, by = c("region", "period")) %>%
+    rbind(hddcdd) %>%
+    pivot_wider(names_from = "variable", values_from = "value")
+
+
+
+  ## Make linear regression to fill historical data ====
+
+  # Assumed minimum u-value
+  minU <- minFactor * min(dataEstimate$uvalue)
+
+  # Prepare data
+  dataEstimate <- dataEstimate %>%
+    mutate(uTrans = log(.data$uvalue - minU))
+
+  # Make fit model
+  model <- lmer("uTrans ~ HDD * log(gdppop + 1) + CDD * log(gdppop + 1) + (1 | region)",
```

Comment:
I never worked with `lmer`. Why don't we have the individual variables `HDD`, `CDD`, and `log(gdppop+ 1)` here?